cdap-docs/build.sh

```diff
@@ -38,227 +66,197 @@ function set_project_path() {
-  echo ""
-  echo "  with"
-  echo "    source         Path to ${PROJECT} source, if not ${PROJECT_PATH}"
-  echo "    test_includes  local, remote or neither (default: remote); must specify source if used"
-  echo ""
-}
-
 function run_command() {
   case "${1}" in
-    all )               build_all;;
-    clean )             clean_builds;;
+    docs-all )          build_all;;
+    docs )              build_docs ${DOCS};;
+    docs-github-only )  build_docs ${GITHUB_ONLY};;
+    docs-web-only )     build_docs ${WEB_ONLY};;
+    docs-github )       build_docs ${GITHUB};;
+    docs-web )          build_docs ${WEB};;
+
+    docs-first-pass )   build_docs_first_pass ;;
+    docs-github-part )  build_docs_github_part;;
+    docs-web-part )     build_docs_web_part;;
+    
+    clean )             clean_all_targets;;
+    javadocs )          build_javadocs;;
+    licenses )          build_license_dependency_pdfs;;
+    sdk )               build_standalone;;
+    version )           print_version;;
+    doc-test )          doc_test;;
+    * )                 usage;;
     doc )               build_docs_outer_level;;
-    docs )              build_docs;;
-    docs-github-part )  build_docs_github ${ARG_2} ${ARG_3};;
-    docs-github )       build_docs_github ${ARG_2} ${ARG_3};;
-    docs-web-part )     build_docs_web ${ARG_2} ${ARG_3};;
-    docs-web )          build_docs_web ${ARG_2} ${ARG_3}; exit 0;;
-    javadocs )          build_javadocs; exit 0;;
-    licenses )          build_license_depends; exit 0;;
-    sdk )               build_sdk; exit 0;;
-    version )           print_version; exit 0;;
-    test )              test; exit 0;;
-    * )                 usage; exit 0;;
   esac
 }
 
-function clean() {
-  cd ${SCRIPT_PATH}
-  rm -rf ${SCRIPT_PATH}/${BUILD}/*
-  mkdir -p ${SCRIPT_PATH}/${BUILD}/${HTML}
-  mkdir -p ${SCRIPT_PATH}/${BUILD}/${SOURCE}
-  echo "Cleaned ${BUILD} directory"
-  echo ""
-}
-
-function copy_source() {
-  echo "Copying source for ${1} (${2}) ..."
-  cd ${SCRIPT_PATH}
-  mkdir -p ${SCRIPT_PATH}/${BUILD}/${SOURCE}/${1}
-  rewrite ${COMMON_PLACEHOLDER} ${BUILD}/${SOURCE}/${1}/index.rst "<placeholder>" "${2}"
+function build_all() {
+  echo "========================================================"
+  echo "========================================================"
+  echo "Building All: GitHub and Web Docs"
+  echo "--------------------------------------------------------"
   echo ""
-}
-
-function copy_html() {
-  echo "Copying html for ${1}..."
-  cd ${SCRIPT_PATH}
-  rm -rf ${SCRIPT_PATH}/${BUILD}/${HTML}/${1}
-  cp -r ${1}/${BUILD}/${HTML} ${BUILD}/${HTML}/${1}
+  clean_all_targets
+  run_command javadocs 
+  run_command docs-first-pass ${ARG_2} 
+  run_command docs-github-part ${ARG_2} 
+  _stash_github_zip 
+  run_command docs-web-part ${ARG_2} 
+  _restore_github_zip
+  echo "--------------------------------------------------------"
+  _bell "Completed \"build all\""
```

Comment:
Correct. When there's variables, you need to use double quotes for them to be interpreted. Simple strings like these can be single-quoted... or is that not working after being passed to the function?